libvast/src/system/connect_to_node.cpp

```diff
@@ -47,6 +49,7 @@ connect_to_node(scoped_actor& self, const caf::settings& opts) {
   auto str = get_or(opts, "system.endpoint", defaults::system::endpoint);
   if (!parsers::endpoint(str, node_endpoint))
     return make_error(ec::parse_error, "invalid endpoint", str);
+  node_endpoint.port.type(port::tcp);
```

Comment:
I wonder if the `endpoint` parser should assume TCP by default.

Reply:
I've discussed this with @tobim at HQ and we've decided that'd be out of scope for this issue. Do you want me to change that?

Reply:
I'm in full agreement that it's out of scope. I still wanted it to hear your opinion on it, though.

Reply:
> I wonder if the `endpoint` parser should assume TCP by default.

The `endpoint` does not delegate to the `port` parser, but uses `u16` instead.  This is a mismatch with the `endpoint` type, which contains a `vast::port`. I think we should change to `port` parser to make the "/protocol" part optional and delegate to that.
The specific semantic of a CAF endpoint only comes in in this function, it should indeed default to `tcp`, but not overwrite if anything other than `unknown` is set, and manually return an error if something other than `tcp` is requested.

Reply:
The `endpoint` parser was actually correct, I missed the alternative path using the `port` parser.